libs/verification-common/src/blueprint_contracts.rs

```diff
@@ -0,0 +1,101 @@
+use bytes::Bytes;
+
+// ERC5202_PREFIX = b"\xFE\x71\x00"  # default prefix from ERC-5202
+// def blueprint_bytecode(self) -> bytes:
+//     blueprint_bytecode = ERC5202_PREFIX + self.bytecode
+//
+//     # the length of the deployed code in bytes
+//     len_bytes = len(blueprint_bytecode).to_bytes(2, "big")
+//     deploy_bytecode = b"\x61" + len_bytes + b"\x3d\x81\x60\x0a\x3d\x39\xf3"
+//
+//     return deploy_bytecode + blueprint_bytecode
+
+pub fn from_runtime_code(code: Bytes) -> Option<Bytes> {
+    let prefix = [0xfe, 0x71, 0x00];
+    code.starts_with(&prefix)
+        .then(|| code.slice(prefix.len()..))
+}
+
+pub fn from_creation_code(code: Bytes) -> Option<Bytes> {
+    if code.len() < 10 {
+        return None;
+    }
+
+    let deploy_bytecode_prefix = [
+        0x61, code[1], code[2], 0x3d, 0x81, 0x60, 0x0a, 0x3d, 0x39, 0xf3,
+    ];
+    if code.starts_with(&deploy_bytecode_prefix) {
+        let len_bytes = code[1] as usize * 256 + code[2] as usize;
+        let blueprint_bytecode = code.slice(deploy_bytecode_prefix.len()..);
+        if blueprint_bytecode.len() == len_bytes {
+            return from_runtime_code(blueprint_bytecode);
+        }
+    }
+
+    None
+}
+
+#[cfg(test)]
+mod tests {
+    use super::*;
+
+    fn from_hex(value: &str) -> Bytes {
+        let bytes: Vec<u8> = hex::FromHex::from_hex(value).unwrap();
+        Bytes::from(bytes)
+    }
+
+    #[test]
+    fn check_blueprint_runtime_code_success() {
+        // A basic check with not very long bytecode
+        let code = from_hex("fe71003461002457602061009a5f395f5160015560015f5561005f61002860003961005f6000f35b5f80fd5f3560e01c60026001821660011b61005b01601e395f51565b63158ef93e81186100535734610057575f5460405260206040f3610053565b633fa4f245811861005357346100575760015460405260206040f35b5f5ffd5b5f80fd0018003784185f810400a16576797065728300030a0013");
+        let expected_parsed_code = from_hex("3461002457602061009a5f395f5160015560015f5561005f61002860003961005f6000f35b5f80fd5f3560e01c60026001821660011b61005b01601e395f51565b63158ef93e81186100535734610057575f5460405260206040f3610053565b633fa4f245811861005357346100575760015460405260206040f35b5f5ffd5b5f80fd0018003784185f810400a16576797065728300030a0013");
+        assert_eq!(
+            Some(expected_parsed_code),
+            from_runtime_code(code),
+            "length encoded as 1 byte"
+        );
+
+        // A bytecode with length encoded as more than 1 byte
+        // https://etherscan.io/address/0x79D584d2D49eC8CE8Ea379d69364b700bd35874D#code
```

Comment:
https://eth.blockscout.com/address/0x79D584d2D49eC8CE8Ea379d69364b700bd35874D?tab=contract

😄

Reply:
It is not and cannot be verified until that PR is merged :(